daemons/controld/controld_control.c

```diff
@@ -549,64 +571,90 @@ static pe_cluster_option crmd_opts[] = {
+            "the presence of a bug."
+    },
+    {
+        "transition-delay", "crmd-transition-delay", "time", NULL,
+        "0s", pcmk__valid_timer,
+        "*** Advanced Use Only *** Enabling this option will slow down "
+            "cluster recovery under all conditions",
+        "Delay cluster recovery for this much time to allow for additional "
+            "events to occur. Useful if your configuration is sensitive to "
+            "the order in which ping updates arrive."
+    },
+    {
+        "stonith-watchdog-timeout", NULL, "time", NULL,
+        NULL, pcmk__valid_sbd_timeout,
+        "How long to wait before we can assume nodes are safely down "
+            "when sbd is in use",
+        NULL
+    },
+    {
+        "stonith-max-attempts", NULL, "integer", NULL,
+        "10", pcmk__valid_positive_number,
+        "How many times fencing can fail before it will no longer be "
+            "immediately re-attempted on a target"
     },
-	{ "stonith-watchdog-timeout", NULL, "time", NULL, NULL, &check_sbd_timeout,
-	  "How long to wait before we can assume nodes are safely down", NULL
-        },
-        { "stonith-max-attempts",NULL,"integer",NULL,"10",&check_positive_number,
-          "How many times stonith can fail before it will no longer be attempted on a target"
-        },   
 
     // Already documented in libpe_status (other values must be kept identical)
-	{ "no-quorum-policy", NULL, "enum", "stop, freeze, ignore, suicide", "stop", &check_quorum, NULL, NULL },
-    { XML_CONFIG_ATTR_SHUTDOWN_LOCK, NULL, "boolean", NULL, "false", &check_boolean, NULL, NULL },
+    {
+        "no-quorum-policy", NULL, "enum", "stop, freeze, ignore, suicide",
+        "stop", pcmk__valid_quorum, NULL, NULL
+    },
+    {
+        XML_CONFIG_ATTR_SHUTDOWN_LOCK, NULL, "boolean", NULL,
+        "false", pcmk__valid_boolean, NULL, NULL
+    },
 };
-/* *INDENT-ON* */
 
 void
 crmd_metadata(void)
 {
-    config_metadata("pacemaker-controld", "1.0",
-                    "controller properties",
-                    "Cluster properties used by Pacemaker's controller,"
-                    " formerly known as crmd",
-                    crmd_opts, DIMOF(crmd_opts));
+    pcmk__print_option_metadata("pacemaker-controld", "1.0",
+                                "Pacemaker controller options",
+                                "Cluster options used by Pacemaker's "
+                                    "controller (formerly called crmd)",
+                                crmd_opts, DIMOF(crmd_opts));
 }
 
 static void
 verify_crmd_options(GHashTable * options)
 {
-    verify_all_options(options, crmd_opts, DIMOF(crmd_opts));
+    pcmk__validate_cluster_options(options, crmd_opts, DIMOF(crmd_opts));
 }
 
 static const char *
 crmd_pref(GHashTable * options, const char *name)
 {
-    return get_cluster_pref(options, crmd_opts, DIMOF(crmd_opts), name);
+    return pcmk__cluster_option(options, crmd_opts, DIMOF(crmd_opts), name);
```

Comment:
It'd be nice if the lists to pcmk__print_option_metadata, pcmk__validate_cluster_options, and pcmk__cluster_option were NULL terminated so we didn't also have to pass the list length in.

Reply:
That's a good idea, but I'll leave it for a separate PR